src/ensembl/src/content/app/browser/Browser.tsx

```diff
@@ -164,116 +193,78 @@ export const Browser: FunctionComponent<BrowserProps> = (
           changeSelectedSpecies(committedSpecies[0].genome_id);
```

Comment:
Here and in the if block above, if we use `return`, we can prevent reading from the localStorage twice.

Reply:
I am actually not sure what `setTrackStatesFromStorage` is doing here 😄 This useEffect now depends on props.match.params.genomeId and props.location.search; while setTrackStatesFromStorage should depend on `genomeId` and `activeEnsObjectId`? I'll move it in its own useEffect.